redfish-core/lib/oem/ibm/pcie_topology_refresh.hpp

```diff
@@ -0,0 +1,154 @@
+#pragma once
+
+#include "async_resp.hpp"
+#include "dbus_utility.hpp"
+#include "redfish_util.hpp"
+
+#include <memory>
+#include <string>
+#include <variant>
+
+namespace redfish
+{
+// NOLINTBEGIN(cppcoreguidelines-avoid-non-const-global-variables)
+// Timer for PCIe Topology Refresh
+static std::unique_ptr<boost::asio::steady_timer> pcieTopologyRefreshTimer;
+static uint countPCIeTopologyRefresh = 0;
+// NOLINTEND(cppcoreguidelines-avoid-non-const-global-variables)
+
+/**
+ * @brief PCIe Topology Refresh monitor. which block incoming request
+ *
+ * @param[in] timer       pointer to steady timer which block call
+ * @param[in] asyncResp   Shared pointer for generating response message.
+ * @param[in] countPCIeTopologyRefresh   how many time this method get called.
+ *
+ * @return None.
+ */
+static void pcieTopologyRefreshWatchdog(
+    const boost::system::error_code& ec, boost::asio::steady_timer* timer,
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp, uint* countPtr)
+{
+    if (ec)
+    {
+        BMCWEB_LOG_ERROR("steady_timer error {}", ec.value());
+        messages::internalError(asyncResp->res);
+        pcieTopologyRefreshTimer = nullptr;
+        (*countPtr) = 0;
+        return;
+    }
+    //  This method can block incoming requests max for 8 seconds. So, each
+    //  call to this method adds a 1-second block, and the maximum call allow is
+    //  8 times which makes it a total of ~8 seconds
+    if ((*countPtr) >= 8)
+    {
+        messages::internalError(asyncResp->res);
+        pcieTopologyRefreshTimer = nullptr;
+        (*countPtr) = 0;
+        return;
+    }
+    ++(*countPtr);
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, "xyz.openbmc_project.PLDM",
+        "/xyz/openbmc_project/pldm", "com.ibm.PLDM.PCIeTopology",
+        "PCIeTopologyRefresh",
+        [asyncResp, timer, countPtr](const boost::system::error_code& ec1,
+                                     const bool pcieRefreshValue) {
+        if (ec1)
+        {
+            BMCWEB_LOG_ERROR("DBUS response error {}", ec1.value());
+            messages::internalError(asyncResp->res);
+            pcieTopologyRefreshTimer = nullptr;
+            (*countPtr) = 0;
+            return;
+        }
+        // After PCIe Topology Refresh, it sets the pcieRefreshValuePtr
+        // value to false. if a value is not false, extend the time, and if
+        // it is false, delete the timer and reset the counter
+        if (pcieRefreshValue)
+        {
+            BMCWEB_LOG_ERROR("pcieRefreshValuePtr time extended");
```

Comment:
Should this be an ERROR? Seems like either WARNING or INFO is more appropriate, or maybe just DEBUG.

Reply:
I think INFO seems better